File: Python/Query.py
import socket

class JCMPQuery:

	def __init__(self, address, port = 7777):
		self.address = address
		self.port = port
		self.update()

	def update(self):
		sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
		error = False
		try:
			sock.settimeout(2)
			sdata = b'\xff\xff\xff\xff\x54\x53\x6f\x75\x72\x63\x65\x20\x45\x6e\x67\x69\x6e\x65\x20\x51\x75\x65\x72\x79\0'
			sock.sendto(sdata, (self.address, self.port))
			rdata = sock.recv(256)[6:].split('\0', 3)
			pi = rdata[1][9:].split('/', 2)
			self.hostname = rdata[0]
			self.players = int(pi[0])
			self.max_players = int(pi[1])
		except Exception:
			error = True

		sock.close()
		self.error = error
		return error

	def isErroneous(self):
		return self.error

	def getHostName(self):
		return self.hostname

	def getPlayers(self):
		return self.players

	def getMaxPlayers(self):
		return self.max_players

File: Perl/Query.pl
package JCMPQuery;

use strict;
use IO::Socket;

sub new
{
	my $class = shift;
	my $self =
	{
		address	=> shift,
		port 	=> shift,
		hostname => "",
		players => -1,
		max_players => -1,
		error => 1
	};
	bless $self, $class;
	$self->update();
	return $self;
}

sub update
{
	my($self) = @_;
	my $sock = IO::Socket::INET->new(Proto => 'udp', PeerPort => $self->{port}, PeerAddr => $self->{address}, Timeout => 2) or return($self->{error} = 1);
	my $sdata = pack("C*", 
		0xff,0xff,0xff,0xff,0x54,
		0x53,0x6f,0x75,0x72,0x63,
		0x65,0x20,0x45,0x6e,0x67,
		0x69,0x6e,0x65,0x20,0x51,
		0x75,0x65,0x72,0x79,0x00);
	my $rawdata;
	$sock->send($sdata) or return($self->{error} = 1);
	$sock->recv($rawdata, 256) or return($self->{error} = 1);
	my @data = split('\x00', substr($rawdata, 6));
	my @pi = split('/', substr($data[1], 9));
	$self->{hostname} = $data[0];
	$self->{players} = $pi[0];
	$self->{max_players} = $pi[1];
	return($self->{error} = 0);
}

sub isErroneous()
{
	my($self) = @_;
	return $self->{error};
}

sub getHostName()
{
	my($self) = @_;
	return $self->{hostname};
}

sub getPlayers()
{
	my($self) = @_;
	return $self->{players};
}

sub getMaxPlayers()
{
	my($self) = @_;
	return $self->{max_players};
}
